Structure mc documentation

jonaslandsgesell opened the pull request:
Fixes #3309

* correctly indent the Reaction Ensemble subsections in the Sphinx docs
* add a note explaining particle ids may be invalidated when a chemical reaction is added

jonaslandsgesell committed:
change indentation and chapter title

jonaslandsgesell committed:
add a word on invalidated particle ids

RudolfWeeber commented:
I think the correct description would be
the particles are no longer numbered contiguously?
Iterators along the lines of
for p in system.part:
 ....
would still work, wouldn.t they?

jonaslandsgesell commented:
Yes iterating across system.part would work, also iterating over valid pids (given in system.part.id) would work.

jonaslandsgesell committed:
rm backward reaction

jngrad committed:
Fix grammar and capitalization

jngrad committed:
Update docs on Reaction Ensemble

Move links to scripts and classes after the algorithm descriptions.
Provide a link to the class instead of a link to the module. Remove
outdated comments. Discuss particle ordering after a reaction step.

jngrad committed:
Apply code review feedback

jngrad commented:
bors r+
